[BUG]

![image](https://github.com/toppev/strike-tab/assets/160243406/eb2ef659-70af-4789-b4c5-86cf2adcadd3)

[22:30:56 WARN]: org.apache.commons.lang.UnhandledException: Plugin StrikeTab v0.3.13 generated an exception while executing task 73
        at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57)
        at org.github.paperspigot.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:23)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NoClassDefFoundError: us/myles/ViaVersion/api/Via
        at ga.strikepractice.striketab.util.ClientSupportKt.getViaVersionVersion(ClientSupport.kt:29)
        at ga.strikepractice.striketab.util.ClientSupportKt.isLegacyClient(ClientSupport.kt:21)
        at ga.strikepractice.striketab.updater.TabbedTabUpdater$onJoin$1.run(TabbedTabUpdater.kt:167)
        at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftTask.run(CraftTask.java:59)
        at org.bukkit.craftbukkit.v1_8_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:53)
        ... 4 more
Caused by: java.lang.ClassNotFoundException: us.myles.ViaVersion.api.Via
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:102)
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:87)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:593)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        ... 9 more

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,6 +14,7 @@ version = "0.3.13-SNAPSHOT"
 repositories {
     jcenter()
     mavenCentral()
+    maven { url = uri("https://repo.viaversion.com") }
     maven { url = uri("https://maven.toppe.dev/repo") }
     maven { url = uri("https://hub.spigotmc.org/nexus/content/repositories/snapshots/") }
     maven { url = uri("https://oss.sonatype.org/content/repositories/snapshots/") }
@@ -33,7 +34,7 @@ dependencies {
     compileOnly("me.clip:placeholderapi:2.11.3")
     compileOnly("net.citizensnpcs:citizensapi:2.0.30-SNAPSHOT")
     compileOnly("com.comphenix.protocol:ProtocolLib:5.1.0")
-    compileOnly("us.myles:viaversion:viaversion")
+    compileOnly("com.viaversion:viaversion-api:5.0.0")
     // If tabbed is updated, might need to trigger a new build on jitpack:
     // https://jitpack.io/#toppev/tabbed/master-SNAPSHOT
     implementation("com.github.toppev:tabbed:master-SNAPSHOT")
@@ -49,7 +50,7 @@ tasks {
     // Download the jar so we don't need any NMS dependencies (there's no API repository)
     register("download-viaversion") {
         val dest = "./libs/viaversion.jar"
-        val source = "https://repo.viaversion.com/everything/us/myles/viaversion/3.2.1/viaversion-3.2.1.jar"
+        val source = "https://repo.viaversion.com/everything/com/viaversion/viaversion/5.0.0/viaversion-5.0.0.jar"
         if (!File(dest).exists()) {
             download(source, dest)
         }

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -9,7 +9,7 @@ plugins {
 }
 
 group = "ga.strikepractice.striketab"
-version = "0.3.13-SNAPSHOT"
+version = "0.3.14-SNAPSHOT"
 
 repositories {
     jcenter()

--- a/src/main/kotlin/util/ClientSupport.kt
+++ b/src/main/kotlin/util/ClientSupport.kt
@@ -5,7 +5,7 @@ import ga.strikepractice.striketab.PREFIX
 import ga.strikepractice.striketab.debug
 import org.bukkit.Bukkit
 import org.bukkit.entity.Player
-import us.myles.ViaVersion.api.Via
+import com.viaversion.viaversion.api.Via
 
 var LEGACY_SUPPORT = true
 

--- a/version.json
+++ b/version.json
@@ -1,6 +1,6 @@
 {
   "README": "This file is used to check for updates and the changelog",
-  "version": "0.3.13",
+  "version": "0.3.14",
   "message-v0.3.4": "Fixed a memory leak in 1.7 tabs",
   "message-v0.3.5": "Minor fixes & tweaks. Added bStats and skin cache for offline players.",
   "message-v0.3.6": "Minor fixes.",
@@ -10,5 +10,6 @@
   "message-v0.3.10": "Some kind of 1.19+ and ProtocolLib 5.x support.",
   "message-v0.3.11": "Fix issues with < 1.19 servers (since 0.3.10)",
   "message-v0.3.12": "Hex color support",
-  "message-v0.3.13": "Better support for newer mc servers"
+  "message-v0.3.13": "Better support for newer mc servers",
+  "message-v0.3.14": "Fix new ViaVersion support"
 }

